Porting/updateAUTHORS.pl

```diff
@@ -0,0 +1,639 @@
+#!/usr/bin/env perl
+package Porting::updateAUTHORS;
+use strict;
+use warnings;
+use Getopt::Long qw(GetOptions);
+use Pod::Usage qw(pod2usage);
+use Data::Dumper;
+use Encode qw(encode_utf8 decode_utf8 decode);
+
+# Info and config for passing to git log.
+#   %an: author name
+#   %aN: author name (respecting .mailmap, see git-shortlog(1) or git-blame(1))
+#   %ae: author email
+#   %aE: author email (respecting .mailmap, see git-shortlog(1) or git-blame(1))
+#   %cn: committer name
+#   %cN: committer name (respecting .mailmap, see git-shortlog(1) or git-blame(1))
+#   %ce: committer email
+#   %cE: committer email (respecting .mailmap, see git-shortlog(1) or git-blame(1))
+#   %H: commit hash
+#   %h: abbreviated commit hash
+#   %s: subject
+#   %x00: print a byte from a hex code
+
+my %field_spec = (
+    "an" => "author_name",
+    "aN" => "author_name_mm",
+    "ae" => "author_email",
+    "aE" => "author_email_mm",
+    "cn" => "committer_name",
+    "cN" => "committer_name_mm",
+    "ce" => "committer_email",
+    "cE" => "committer_email_mm",
+    "H"  => "commit_hash",
+    "h"  => "abbrev_hash",
+    "s"  => "commit_subject",
+);
+
+my @field_codes = sort keys %field_spec;
+my @field_names = map { $field_spec{$_} } @field_codes;
+my $tformat= join "%x00", map { "%" . $_ } @field_codes;
+
+sub _make_name_author_info {
+    my ($author_info, $commit_info, $name_key)=@_;
+    (my $email_key= $name_key)=~s/name/email/;
+    my $email= $commit_info->{$email_key};
+    my $name= $commit_info->{$name_key};
+
+    my $line = $author_info->{"email2line"}{$email}
+            // $author_info->{"name2line"}{$name};
+
+    $line //= sprintf "%-31s<%s>",
+        $commit_info->{$name_key}, $commit_info->{$email_key};
+    return $line;
+}
+
+sub _make_name_simple {
+    my ($commit_info, $key)= @_;
```

Comment:
consider tidy the file?

Reply:
Is there something specific you object to? I left hug my assignment operators deliberately, and as long as there is no standard format for our scripts and this is a new script I don't see why this script shouldn't use my style. The rule is "new scripts Author chooses the style, otherwise follow the style of the file".  I can run perltidy with my favorite options and include the perltidy specifier in the file however if you like, but AFAIK it won't change that line of code. :-)

I raised the idea of having a standard perltidy format as in the name of consistency I would be willing to give up my style eccentricities if others give up theirs, but it was not well received.

Reply:
I ran perltidy on it, and included my perltidy options in the script. I couldn't decide what was best tho, putting it in the pod so it can be easily C&P executed, or putting it in a comment. For now I put it in a comment at the top of the file. Thanks for the feedback, as much as I like my style the file wasn't consistent, and now it is. Thanks for the feedback!

Reply:
thanks for considering it, I know we do not have a good tidy policy in the whole codebase for C, Perl...